src/promptflow-evals/promptflow/evals/evaluators/_relevance/_relevance.py

```diff
@@ -44,7 +48,13 @@ def __init__(self, model_config: AzureOpenAIModelConfiguration):
         if model_config.api_version is None:
             model_config.api_version = "2024-02-15-preview"
 
-        prompty_model_config = {"configuration": model_config}
+        prompty_model_config = {
+            "configuration": model_config,
+        }
+
+        prompty_model_config.update({"parameters": {"extra-headers": {"x-ms-user-agent": USER_AGENT}}})\
```

Comment:
If prompty_model_config["parameters"]["extra-headers"] has a value, the original value will be overwritten

Reply:
Agreed however the prompty used in this evaluator is not a user defined

Reply:
Prompty here is a part of relevance evaluator and it does not have this extra_headers param